doozerlib/cli/__main__.py

```diff
@@ -1518,6 +1522,132 @@ def config_read_group(runtime, key, as_len, as_yaml, permit_missing_group, defau
+    """
+
+    content = get_releases(runtime.group)
+
+    if as_len:
+        output = len(content['releases'])
+    elif as_yaml:
+        output = yaml.safe_dump(content)
+    else:
+        output = content
+
+    if out_file:
+        # just in case
+        out_file = os.path.expanduser(out_file)
+        try:
+            with io.open(out_file, 'w', encoding="utf-8") as f:
+                f.write(str(output))
+        except PermissionError:
+            click.echo(f'Permission denied: could not write to {out_file}')
+            sys.exit(1)
+
+    else:
+        click.echo(output)
+
+
+@cli.command("config:read-assemblies", short_help="Output aspects of assemblies defined in releases.yml")
+@click.option("--assembly", help="Group assembly to analyze", required=True)
+@click.option("--default", help="Value to print if key cannot be found", default=None)
+@click.option("--length", "as_len", default=False, is_flag=True, help='Print length of dict/list specified by key')
+@click.option("--yaml", "as_yaml", default=False, is_flag=True, help='Format output as YAML')
+@click.option("--out-file", help="Output contents to a file instead of stdout", default=None)
+@click.argument("key", nargs=1, metavar="KEY", type=click.STRING, default=None, required=False)
+@pass_runtime
+def config_read_assemblies(runtime, assembly, default, as_len, as_yaml, out_file, key):
+    """
+    Read data from releases.yaml for given group, assembly and key.
+    An assembly must be specified. To get a global representation of release.yaml,
+    use doozer config:read-releases instead
+
+    Usage:
+
+    $ doozer --group=openshift-4.13 config:read-assemblies --assembly 4.13.1
+
+    $ doozer --group=openshift-4.13 config:read-assemblies --assembly 4.13.1 --yaml
+
+    $ doozer --group=openshift-4.13 config:read-assemblies --assembly 4.13.1 --yaml --out-file /tmp/out.yaml
+
+    $ doozer --group=openshift-4.13 config:read-assemblies --assembly 4.13.1 --yaml assembly.issues.exclude --length
+
+    $ doozer --group=openshift-4.13 config:read-assemblies --assembly 4.13.1 --yaml assembly.promotion_permits
+
+    $ doozer --group=openshift-4.13 config:read-assemblies --assembly 4.13.2 --yaml assembly.promotion_permits --default []
+    """
+
+    runtime.initialize(**CONFIG_RUNTIME_OPTS)
+    releases = get_releases(runtime.group)['releases']
+    assembly_data = releases[assembly]
+
+    if key is not None:
+        assembly_data = dict_get(assembly_data, key, None)
+        if assembly_data is None:
+            if default is not None:
+                click.echo(default)
+                sys.exit(0)
+            raise DoozerFatalError('No default specified and unable to find key: {}'.format(key))
+
+    if as_len:
+        output = len(assembly_data)
+
+    elif as_yaml:
+        output = yaml.safe_dump(assembly_data)
+
+    else:
+        output = assembly_data
+
+    if out_file:
+        # just in case
+        out_file = os.path.expanduser(out_file)
```

Comment:
same here.

Reply:
Done in 9fb059b539e870c7a5b3fafe9e5543e21030a5ff